From b723ac5dbf5d055a614fdb8e2b755c086e47c086 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Thu, 23 May 2024 10:56:12 +0200
Subject: [PATCH 6/9] qmicli-dms: do not leak result on error

The result is not properly disposed in error handling path, resulting
in a potential leak.
---
 src/qmicli/qmicli-dms.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qmicli/qmicli-dms.c b/src/qmicli/qmicli-dms.c
index d001656f..afba1e19 100644
--- a/src/qmicli/qmicli-dms.c
+++ b/src/qmicli/qmicli-dms.c
@@ -3136,6 +3136,7 @@ get_stored_image_list_stored_images_ready (QmiClientDms *client,
             qmi_message_dms_list_stored_images_output_unref (output);
             g_object_unref (task);
             operation_shutdown (FALSE);
+            get_stored_image_result_free (result);
             return;
         }
 
-- 
2.45.2

